digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/RoadAddressMapper.scala

```diff
@@ -205,6 +205,23 @@ trait RoadAddressMapper {
       geom1.last.distance2DTo(geom2.head) + geom1.head.distance2DTo(geom2.last))
   }
 
+  /**
+    * Measure summed distance between two geometries: head-to-head + tail-to-head vs. head-to-tail + tail-to-head
+    * The measurement is taken after the geometries are reduced to the origin point.
+    * @param geom1 Geometry 1
+    * @param geom2 Goemetry 2
+    * @return h2h distance, h2t distance sums
+    */
+
+  def distancesBetweenEndPointsInOrigin(geom1: Seq[Point], geom2: Seq[Point]): (Double, Double) = {
+    val sourcePoint = geom1.minBy(p => p.distance2DTo(Point(0,0,0)))
```

Comment:
You can create a method to transfer the geometry to start at some point, at geometry utils.
`transferGeometry(geom: Seq[Point], to: Point)`